crypto/ec/curve448/curve448utils.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017-2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2015 Cryptography Research, Inc.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * Originally written by Mike Hamburg
+ */
+
+#ifndef __C448_COMMON_H__
+# define __C448_COMMON_H__ 1
+
+# include <openssl/e_os2.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * Internal word types. Somewhat tricky.  This could be decided separately per
+ * platform.  However, the structs do need to be all the same size and
+ * alignment on a given platform to support dynamic linking, since even if you
+ * header was built with eg arch_neon, you might end up linking a library built
+ * with arch_arm32.
+ */
+# ifndef C448_WORD_BITS
+#  if (defined(__ILP64__) || defined(__amd64__) || defined(__x86_64__) \
+      || (((__UINT_FAST32_MAX__)>>30)>>30))
```

Comment:
any idea why we are shifting twice?
why 60 bits, and not 63?

Reply:
I'm guessing to avoid a warning if it's 32 bit?

Reply:
Don't know. Maybe @bitwiseshiftleft can enlighten us.

Reply:
This is not working for non-GNU (or non-GNU-alikes) or elderly GNU compilers. Well, not this #if specifically, but what it enables, __int128 few lines below. I'm not really clear on which type is officially supported as builtin in long term, [and there were some changes,] but for poly1305 I've chosen __int128 alone, i.e. `unsigned __int128` and not __uint128_t. Well, it doesn't mean it's **the one**, but it served well so far. As for pre-processor condition poly1305 uses
```
# if !defined(PEDANTIC) && \
     (defined(__SIZEOF_INT128__) && __SIZEOF_INT128__==16) && \
     (defined(__SIZEOF_LONG__) && __SIZEOF_LONG__==8)
```
with rationale that it filters out non-GNU (and non-alikes) as well as elderly GNU.

Just in case, "not working" means "actually tried and confirmed".

Reply:
Either use @dot-asm's suggestion for the preprocessor condition or
just take this part out for the moment, C448_WORD_BITS = 32 should always work.

Reply:
I think we change this condition to be simply:
```
# if defined(__SIZEOF_INT128__) && __SIZEOF_INT128__==16
```

what do you think?

Reply:
Works for me. Done.